### PR TITLE
feat(#249): KtFieldNumber Enhancements

### PR DESCRIPTION
**Pull request by carsoli**
Preliminary Steps before introducing `KtFieldCurrency`

As per description of #249, this is what is covered in this PR's scope:
- [x] Extend `KtFieldNumber` First:
  - [x] Support parsing all 3 decimal separators as one
  - [x] Support decimal separator through translation context
  - [x] Remove the 3 decimal places cap on the number field 
     - Make it customizable through a prop - and default to `3`

### CODE REVIEW

File: packages/kotti-ui/source/kotti-i18n/types.ts [hunk 1]
```diff
@@ -31,6 +37,7 @@ export namespace KottiI18n {
 	export type Props = {
 		locale: SupportedLanguages
 		messages: DeepPartial<Messages>
+		numberFormat: NumberFormat
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-i18n/types.ts:40]: Actually, we should probably allow this to be a DeepPartial for future extensibility

File: packages/kotti-ui/source/kotti-field-number/constants.ts [hunk 2]
```diff
@@ -23,22 +25,28 @@ export const KOTTI_FIELD_NUMBER_PROPS = {
...
 
-export const DECIMAL_PLACES = 3
-// FIXME: should we consider manually setting the decimal separator using the translation context
-export const DECIMAL_SEPARATOR = (1.1).toLocaleString().replace(/\d/g, '')
+/**
+ * RegExp character set for use within other regular expressions
+ */
+export const DECIMAL_SEPARATORS_CHARACTER_SET = [
+	'[',
+	...Object.values(DecimalSeparator).map((x) => `\\${x}`),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-field-number/constants.ts:33]: I'm not sure if escaping is necessary in character sets

**Review by FlorianWendelborn — APPROVED** [review]

**Comment by HansAuger** [comment]
Not a blocker, but it would make sense to add single select field to `packages/documentation/pages/usage/components/form.vue` another select field to pick a comma separator net to the language. Right now there is no number field on that page to see the effect though, that would need to be added a well.

File: packages/kotti-ui/source/kotti-field-number/constants.ts (hunk 2 above)
Comment (carsoli) [reply packages/kotti-ui/source/kotti-field-number/constants.ts]: changed in latest

File: packages/kotti-ui/source/kotti-i18n/KtI18nContext.ts [hunk 3]
```diff
@@ -8,11 +10,18 @@ const KtI18nContext = defineComponent({
 	props: {
 		locale: { required: true, type: String },
 		messages: { default: () => ({}), type: Object },
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-i18n/KtI18nContext.ts:12]: : KottiI18n.Messages

**Comment by HansAuger** [comment]
LGTM, it behaves a lot better than before.

One thing that still happens is that if you enter a number like `1.001` and then delete the last digit,  you don't get `1.00` but `1`. If you think we can deal with that scenario in the scope of this PR, that would be amazing to fix. But also not a blocker, since this behavior already existed before and all other problems seem to be fixed.

**Review by HansAuger — APPROVED** [review]

**Comment by HansAuger** [comment]
@carsoli The only issue I have is that we then do not document this feature at all. Not a super big deal, we also don't document the `messages` prop of `KtI18NContext for example. But something to keep in mind.

**Comment by carsoli** [comment]
> One thing that still happens is that if you enter a number like 1.001 and then delete the last digit, you don't get 1.00 but 1. If you think we can deal with that scenario in the scope of this PR, that would be amazing to fix. But also not a blocker, since this behavior already existed before and all other problems seem to be fixed.

@HansAuger the `.00` would be handled within the currency split - can be exposed as an extra prop IMO but now that you mention it, let me check if it was on the list of tasks just so we don't forget.

Merging for now.

**Pull request merged**